demo_nodes_cpp/src/parameters/test_param_logic.cpp

```diff
@@ -0,0 +1,104 @@
+//
+// Created by deepanshu on 6/1/22.
+//
+
+#include "string"
+#include "memory"
+
+#include <utility>
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/logging.hpp"
+#include "rcl_interfaces/msg/set_parameters_result.hpp"
+
+// Notes: // failure in one param does not affect second param. --> validation
+
+class ParamTest{
```

Comment:
I think we should rename this class and file to be more consistent with other demos. Maybe `local_parameter_callback.cpp`?